subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -48,66 +40,106 @@ static struct bt_mesh_silvair_enocean_srv *model_resolve(const bt_addr_le_t *add
 	return NULL;
 }
 
-static uint32_t delay_get(struct bt_mesh_model_pub *pub, uint32_t pub_num)
-{
-	return (BT_MESH_PUB_MSG_TOTAL(pub) - pub_num) * 50;
-}
-
-static void send_onoff(struct bt_mesh_silvair_enocean_button *button,
-		       bool on_off)
+static void send_delta(struct bt_mesh_silvair_enocean_button *button, uint32_t delay,
+		       bool new_transaction)
 {
 	struct bt_mesh_model_transition transition;
-	struct bt_mesh_onoff_set msg = {
-		.on_off = on_off,
+	struct bt_mesh_lvl_delta_set msg = {
 		.transition = &transition,
+		.new_transaction = new_transaction,
+		.delta = button->delta
+	};
+	struct bt_mesh_msg_ctx ctx = {
+		.addr = button->onoff.pub.addr,
+		.send_ttl = button->onoff.pub.ttl,
+		.app_idx = button->onoff.pub.key
 	};
 
-	button->onoff.pub.retransmit = BT_MESH_PUB_TRANSMIT(3, 50);
+	if (CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TRANSITION_TIME == -1) {
```

Comment:
We need to keep it to make cases where it is already used compile after this. Since we are putting this commit in a patch release I really don't want to alter the API at all for this, we can make a second commit to remove all uncessary config options (there are several, not only this)